### ISSUE TITLE
Passing an unpacked dictionary and keyword arguments to `**kwargs` gets unexpected errors

### ISSUE THREAD

**Issue by hyperkai** [python/mypy#20650]
*Memo:
- mypy test.py
- mypy 1.19.1
- Python 3.14.0

Passing an unpacked list and positional arguments to `*args: *tuple[int, ...]` doesn't get error as shown below:

```python
def func(*args: *tuple[int, ...]) -> None: ...

func(*[0, 1, 2]) # No error
func(0, 1, 2)    # No error
```

But, passing an unpacked dictionary and keyword arguments to `**kwargs: dict[str, int]` gets the errors as shown below:

```python
def func(**kwargs: dict[str, int]) -> None: ...

func(**{'A': 0, 'B': 1}) # Error
```

> error: Argument 1 to "func" has incompatible type "**dict[str, int]"; expected "dict[str, int]"

> note: "dict" is invariant

> note: Consider using "Mapping" instead, which is covariant in the value type

```python
def func(**kwargs: dict[str, int]) -> None: ...

func(A=0, B=1) # Error
```

> error: Argument "A" to "func" has incompatible type "int"; expected "dict[str, int]"

> error: Argument "B" to "func" has incompatible type "int"; expected "dict[str, int]"

And, passing an unpacked dictionary to `**kwargs: Unpack[Person]` based on [TypedDict](https://docs.python.org/3/library/typing.html#typing.TypedDict) gets the errors while passing keyword arguments to `**kwargs: Unpack[Person]` based on `TypedDict` doesn't get error as shown below:

```python
from typing import TypedDict, Unpack

class Person(TypedDict):
    name: str
    age: int

def func(**kwargs: Unpack[Person]) -> None: ...

func(**{'name': 'John', 'age': 32}) # Error
func(name='John', age=32)           # No error
```

> error: Argument 1 to "func" has incompatible type "**dict[str, object]"; expected "str"

> error: Argument 1 to "func" has incompatible type "**dict[str, object]"; expected "int"

**Comment by A5rocks** [comment]
`**kwargs: dict[str, int]` should be `**kwargs: int`

Unpacking into a TypedDict isn't supported because, well, we don't have more specific dict literals. I think there's an issue open for this?

**Comment by A5rocks** [comment]
TypedDict issue is basically just https://github.com/python/mypy/issues/5382
